Payline/Payline.php

```diff
@@ -169,6 +169,10 @@ public function doRefund($paymentToken, $comment = '', $sequenceNumber = 0)
 
         $paymentDetails['payment']['action'] = WebGatewayInterface::CODE_ACTION_DOREFUND;
 
+        if($amount) {
+            $paymentDetails['payment']['amount'] = $amount;
+        }
```

Comment:
No it should not because it is set by default on the total amount of the transaction via getWebPaymentDetails. Amount parameter should be use for partial refund only.